File: leetcode/2337.移动片段得到字符串.cpp
class Solution {
public:
    bool canChange(string start, string target) {
        //太勾八优雅了
        auto s = start, t = target;
        s.erase(remove(s.begin(), s.end(), '_'), s.end());
        t.erase(remove(t.begin(), t.end(), '_'), t.end());
        if (s != t) return false;
        for (int i = 0, j = 0; i < start.length(); ++i) {
            if (start[i] == '_') continue;
            while (target[j] == '_') ++j;
            if (i != j && (start[i] == 'L') != (i > j)) return false;
            ++j;
        }
        return true;
    }
};

File: leetcode/2337.移动片段得到字符串.py
class Solution:
    def canChange(self, start: str, target: str) -> bool:
        s, t = start.replace('_', ""), target.replace('_', "")
        print(s, t)
        if s != t: return False
        j = 0
        for i in range(len(start)):
            if start[i] == '_': continue
            while target[j] == '_': j += 1
            if start[i] == 'L' and i < j: return False
            if start[i] == 'R' and i > j: return False
            j += 1
        return True
        